modules/wikipedia/index.js

```diff
@@ -2,21 +2,37 @@ const wiki = require('wikijs').default;
 
 module.exports.commands = ['wiki', 'wikipedia'];
 
-function prettyPage(page) {
+const urlRegex = /^(?:https?:\/\/)?(?:en\.)?wikipedia\.org\/wiki\/(.+)/;
+
+const errorMessage = term => `No Wikipedia page found for "${term}"`;
+
+function shortSummary(page, withUrl) {
   return page.summary()
     // Get the first "sentence" (hopefully)
     .then(str => str.substr(0, str.indexOf('.') + 1))
     // Truncate with an ellipsis if length exceeds 250 chars
     .then(str => (str.length > 250 ? `${str.substr(0, 250)}...` : str))
-    // Append Wikipedia URL
-    .then(str => `${str} <${page.raw.canonicalurl}>`);
+    // Append URL if requested
+    .then(str => (withUrl ? `${str} <${page.raw.canonicalurl}>` : str));
 }
 
-module.exports.run = function run(remainder, parts, reply) {
+module.exports.run = (remainder, parts, reply) => {
   wiki().search(remainder, 1)
     .then(data => data.results[0])
     .then(wiki().page)
-    .then(prettyPage)
+    .then(page => shortSummary(page, true))
     .then(reply)
-    .catch(() => reply(`No Wikipedia page found for "${remainder}"`));
+    .catch(() => reply(errorMessage(remainder)));
+};
+
+module.exports.url = (url, reply) => {
+  if (urlRegex.test(url)) {
+    const [, match] = urlRegex.exec(url);
+    const title = decodeURIComponent(match);
+
+    wiki().page(title)
```

Comment:
I think we want to use `findById` since the url fragment is technically an ID not a title.
I don't think there's any case where a search for an id won't give the page, but maybe there's one?

Reply:
Oops, I'm wrong, id is actually the id number of the page.

The documentation for `findById` is wrong, I'll pr that against the wikijs package.